src/Units/Unit.java

```diff
@@ -46,27 +47,35 @@ public Unit(Commander co, UnitModel um)
 		if (model.holdingCapacity > 0)
 			heldUnits = new Vector<Unit>(model.holdingCapacity);
 	}
-	
-	public void initTurn()
-	{
+
+	public void initTurn(Location locus) {
 		isTurnOver = false;
 		fuel -= model.idleFuelBurn;
-		if (captureTarget != null && captureTarget.getResident() != this)
-		{
+		if (captureTarget != null && captureTarget.getResident() != this) {
 			captureTarget = null;
 			captureProgress = 0;
 		}
+		if (HP < 9.99) {
+			if (model.canRepairOn(locus) && locus.getOwner() == CO) {
+				int cost = model.moneyCost * Math.min(10 - getHP(), 2)/10; // 1/10th the cost to repair fully
```

Comment:
I misread the source I got the cost from, and continued in my misunderstanding while using the proper numbers. I think I'm officially crazy.